src/dispatcher.rs

```diff
@@ -428,11 +428,17 @@ impl Dispatcher {
     }
 
     fn on_grpc_receive_initial_metadata(&self, token_id: u32, headers: u32) {
-        let context_id = *self
-            .grpc_streams
-            .borrow_mut()
-            .get(&token_id)
-            .expect("invalid token_id");
+        let grpc_streams_ref = self.grpc_streams.borrow_mut();
+        let context_id_hash_slot = grpc_streams_ref
+            .get(&token_id);
+        let context_id = match context_id_hash_slot {
+            Some(id) => *id,
+            None => {
+                // TODO: change back to a panic once underlying issue is fixed.
```

Comment:
This matches the language in a very similar PR a while back: https://github.com/proxy-wasm/proxy-wasm-rust-sdk/pull/154